cmd/options/controller_options.go

```diff
@@ -27,8 +27,12 @@ type ControllerOptions struct {
 	// ExtraVolumeTags is a map of tags that will be attached to each dynamically provisioned
 	// volume.
 	ExtraVolumeTags map[string]string
+	// ID of the kubernetes cluster. This is used only to create the same tags on volumes that
+	// in-tree volume volume plugin does.
+	ClusterID string
 }
 
 func (s *ControllerOptions) AddFlags(fs *flag.FlagSet) {
 	fs.Var(cliflag.NewMapStringString(&s.ExtraVolumeTags), "extra-volume-tags", "Extra volume tags to attach to each dynamically provisioned volume. It is a comma separated list of key value pairs like '<key1>=<value1>,<key2>=<value2>'")
+	fs.StringVar(&s.ClusterID, "cluster-id", "", "ID of the Kubernetes cluster (optional).")
```

Comment:
I'm afraid `--cluster-id` alone doesn't represent what this flag is really doing (adding labels with cluster ID to the volumes).

Also, I'd make the flag explicitly mention k8s (in addition to the help message already mentioning it). Something like `k8s-cluster-id-volume-tags`. It's similar to the flag above, plus it emphasizes it's k8s-specific.

Reply:
Renamed to `--k8s-tag-cluster-id=` with description "ID of the Kubernetes cluster used for tagging provisioned EBS volumes (optional)"